vm/sdk/src/lib.rs

```diff
@@ -50,7 +50,7 @@ pub(crate) fn bytes_to_u32_vec<T: AsRef<[u8]>>(bytes: T) -> Vec<u32> {
 }
 
 pub trait VerifiableTemplate {
-  fn verify(&self, tx: &[u8], signature: &[u8; 64]) -> bool;
+  fn verify(&self, tx: Vec<u8>, signature: [u8; 64]) -> bool;
```

Comment:
Why change this function signature?

Reply:
Because the decoding from io needs an owned type to decode into. It cannot decode into a reference type.